pkg/image/driver.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package image
+
+import (
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"github.com/golang/glog"
+
+	"github.com/kubernetes-csi/drivers/pkg/csi-common"
+)
+
+type driver struct {
+	csiDriver *csicommon.CSIDriver
+	endpoint  string
+
+	ids *csicommon.DefaultIdentityServer
+	ns  *nodeServer
+
+	cap   []*csi.VolumeCapability_AccessMode
+	cscap []*csi.ControllerServiceCapability
+}
+
+const (
+	driverName = "csi-imageplugin"
+)
+
+var (
+	version = "0.0.1"
+)
+
+func NewDriver(nodeID, endpoint string) *driver {
+	glog.Infof("Driver: %v version: %v", driverName, version)
+
+	d := &driver{}
+
+	d.endpoint = endpoint
+
+	csiDriver := csicommon.NewCSIDriver(driverName, version, nodeID)
+	csiDriver.AddVolumeCapabilityAccessModes([]csi.VolumeCapability_AccessMode_Mode{csi.VolumeCapability_AccessMode_MULTI_NODE_MULTI_WRITER})
+	// NFS plugin does not support ControllerServiceCapability now.
```

Comment:
NFS plugin?

Reply:
I missed that comment. It started life out as the nfs plugin. I'll remove it.